corehq/ex-submodules/casexml/apps/phone/tasks.py

```diff
@@ -92,18 +94,24 @@ def update_celery_state(sender=None, headers=None, **kwargs):
 )
 def prune_synclogs():
     """
-    Drops all partition tables containing data that's older than 63 days (7 weeks)
+    Drops all partition tables containing data that's older than 63 days (9 weeks)
     """
-    oldest_synclog = SyncLogSQL.objects.aggregate(Min('date'))['date__min']
-    while oldest_synclog and (datetime.today() - oldest_synclog).days > SYNCLOG_RETENTION_DAYS:
+    db = router.db_for_write(SyncLogSQL)
+    oldest_date = SyncLogSQL.objects.aggregate(Min('date'))['date__min']
+    while oldest_date and (datetime.today() - oldest_date).days > SYNCLOG_RETENTION_DAYS:
         year, week, _ = oldest_synclog.isocalendar()
         table_name = "{base_name}_y{year}w{week}".format(
             base_name=SyncLogSQL._meta.db_table,
             year=year,
             week="%02d" % week
         )
         drop_query = "DROP TABLE IF EXISTS {}".format(table_name)
-        db = router.db_for_write(SyncLogSQL)
         with connections[db].cursor() as cursor:
             cursor.execute(drop_query)
-        oldest_synclog = SyncLogSQL.objects.aggregate(Min('date'))['date__min']
+        oldest_date += timedelta(weeks=1)
+
+    # find and log synclogs for which the trigger did not function properly
+    with connections[db].cursor() as cursor:
+        cursor.execute("select count(*) from only phone_synclogsql")
+        orphaned_synclogs = cursor.fetchone()[0]
+        metrics_gauge('commcare.orphaned_synclogs', orphaned_synclogs)
```

Comment:
As this PR stands, nothing will actually clean up these orphaned synclogs after 9 weeks, they just won't hold up the rest of the cleanup, right?

Reply:
Right, leaving them around gives us a chance to investigate, especially if it happens again. Rather than immediately destroying the evidence.